transport/tchannel/outbound.go

```diff
@@ -64,6 +64,17 @@ type outbound struct {
 	HostPort string
 }
 
+func (o outbound) Start() error {
+	// TODO: Should we create the connection to HostPort (if specified) here or
```

Comment:
Hmm, I imagine if an application specifies an outbound, it will most likely use it, so connecting earlier might be better. However, I'd also be fine with a "warmup" phase in services that makes a bunch of requests to warm up all the connections etc, before it's officially up and ready to serve traffic.